### ISSUE TITLE
"initialBindings" not applied when using functions

### ISSUE THREAD

**Issue by sdevalk** [comunica/comunica#640]
#### Issue type:

- :bug: Bug

____
#### Description:

I want to use the `initialBindings` option in a query. However, my pre-bound value is not processed when using e.g. a CONTAINS() function:

```javascript
import * as Comunica from '@comunica/actor-init-sparql';
import { LoggerPretty } from '@comunica/logger-pretty';
import { Bindings } from '@comunica/bus-query-operation';
import { literal } from '@rdfjs/data-model';

(async () => {
  const myEngine = Comunica.newEngine();

  const query = `PREFIX schema: <http://schema.org/>
    SELECT *
    WHERE {
      ?s a schema:Person .
      ?s schema:name ?name .
      FILTER(CONTAINS(?name, ?searchTerm))
    }`;

  const config = {
    log: new LoggerPretty({ level: 'debug' }),
    sources: [
      {
        type: 'sparql',
        value: 'https://api.data.netwerkdigitaalerfgoed.nl/datasets/rkd/rkdartists/services/rkdartists/sparql',
      },
    ],
    initialBindings: Bindings({ '?searchTerm': literal('Gogh') }),
  };

  await myEngine.query(query, config);
})();
```

The query generated by Comunica looks like this:

```sparql
SELECT ?s ?name WHERE {
  ?s <http://www.w3.org/1999/02/22-rdf-syntax-ns#type> <http://schema.org/Person>.
  ?s <http://schema.org/name> ?name.
  FILTER(CONTAINS(?name, ?searchTerm))
}
```

Expected result: `"Gogh"` instead of `?searchTerm`.

However, if I execute the query underneath...

```javascript
  const query = `PREFIX schema: <http://schema.org/>
    SELECT *
    WHERE {
      ?s a schema:Person .
      ?s schema:name ?searchTerm
    }`;
```

...then the generated query looks like this:

```sparql
SELECT ?s WHERE {
  ?s <http://www.w3.org/1999/02/22-rdf-syntax-ns#type> <http://schema.org/Person>.
  ?s <http://schema.org/name> "Gogh"^^<http://www.w3.org/2001/XMLSchema#string>.
}
```

____
#### Environment:

| software            | version
| ------------------- | -------
| Comunica Init Actor | 1.10.0
| node                | v12.0.0
| npm                 | 6.9.0
| yarn                | 1.15.2
| Operating System    | linux (Linux 4.9.184-linuxkit)

**Comment by rubensworks** [comment]
Thanks for reporting!

Basically, we'll have to rewrite the [`applyInitialBindings`](https://github.com/comunica/comunica/blob/master/packages/actor-init-sparql/lib/ActorInitSparql-browser.ts#L76-L109) function.
For this, we should use SPARQLAlgebra's `mapOperation`, as is done [here](https://github.com/comunica/comunica/blob/master/packages/actor-query-operation-leftjoin-left-deep/lib/ActorQueryOperationLeftJoinLeftDeep.ts#L60-L83).
@joachimvh I assume `mapOperation` also maps filter expressions recursively?

**Comment by joachimvh** [comment]
> @joachimvh I assume `mapOperation` also maps filter expressions recursively?

It does

**Comment by rubensworks** [comment]
Fix released in 1.11.1.
